src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java

```diff
@@ -130,7 +133,18 @@ private String generateUniqueId() {
     return UUID.randomUUID().toString();
   }
 
-  public void deleteHouseFromCommunityByHouseId(String houseId) {
-    communityHouseRepository.deleteByHouseId(houseId);
+  public boolean removeHouseFromCommunityByHouseId(String communityId, String houseId) {
+    return communityRepository.findByCommunityId(communityId)
+        .map(community -> {
+          CommunityHouse house = communityHouseRepository.findByHouseId(houseId);
+          if(house != null && community.getHouses().contains(house)) {
+            community.getHouses().remove(house);
+            communityRepository.save(community);
+            return true;
```

Comment:
I can't make `communityHouseRepository.findByHouseId(houseId)` return `Optinal<CommunityHouse>` because this change affect on `HouseSDJpaService`. May be should create issue for refactor `HouseSDJpaService`